packages/flame/lib/src/components/component.dart

```diff
@@ -570,16 +599,31 @@ class _LifecycleManager {
   /// finish loading in arbitrary order.
   final Queue<Component> _children = Queue();
 
+  /// Queue for removing children from a component.
+  ///
+  /// Components that were placed into this queue will be removed from [parent]
+  /// when the pending events are resolved.
+  final Queue<Component> _dying = Queue();
```

Comment:
I think this should have a clearer name to better reflect the state that the components are in.
Like always, hard with naming, maybe `_toBeRemoved`, `_removals` or something?

Reply:
I like `_removals`